### PR TITLE
IDENTITY-6654: Unit test coverage for OIDCSession management

### PR DESCRIPTION
**Pull request by piraveena**

### CODE REVIEW

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.wso2.carbon.identity.oidc.session.servlet.TestUtil;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+public class OIDCSessionParticipantCacheTest extends PowerMockTestCase {
+
+    private static final String SESSION_ID= "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java]: Fix formatting in this class.

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+package org.wso2.carbon.identity.oidc.session;
+
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.testng.Assert.*;
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java]: Better to remove wild card imports.

**Comment by mefarazath** [comment]
Note: Use "IDENTITY-<JIRA_ID>" as PR format not "Identity"

**Comment by piraveena** [comment]
Fixed with requested changes

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Test
+    public void testGetInstance() {
+        assertNotNull(OIDCSessionParticipantCache.getInstance());
+    }
+
+    @Test
+    public void testAddToCache() {
+        TestUtil.startTenantFlow("carbon.super");
+        OIDCSessionParticipantCacheKey key=mock(OIDCSessionParticipantCacheKey.class);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java]: Formatting

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        OIDCSessionParticipantCacheEntry entry= mock(OIDCSessionParticipantCacheEntry.class);
+        when(key.getSessionID()).thenReturn(SESSION_ID);
+        OIDCSessionParticipantCache.getInstance().addToCache(key,entry );
+        assertNotNull(OIDCSessionParticipantCache.getInstance().getValueFromCache(key));
+    }
+
+    @Test
+    public void testClearCacheEntry() {
+        TestUtil.startTenantFlow("carbon.super");
+        OIDCSessionParticipantCacheKey key=mock(OIDCSessionParticipantCacheKey.class);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java]: Formatting

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    private static final String SESSION_ID = "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
+
+    @Test
+    public void testGetInstance() {
+        assertNotNull(OIDCSessionParticipantCache.getInstance());
+    }
+
+    @Test
+    public void testAddToCache() {
+        TestUtil.startTenantFlow("carbon.super");
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java]: Use constants from MultiTenantConstants

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        String client_id = "ES9l2uUf8AzNOfmGS9lPEIsdrR8a";
+        oidcSessionState.addSessionParticipant(client_id);
+        Set sessionParticipants = oidcSessionState.getSessionParticipants();
+        assertNotNull(sessionParticipants.contains(client_id), "Client_id is a session participant");
+    }
+
+    @Test
+    public void testSetAuthenticated() throws Exception {
+        oidcSessionState.setAuthenticated(true);
+        assertNotNull(oidcSessionState.isAuthenticated());
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java]: you can use assertTrue()  if oidcSessionState.isAuthenticated() returns a boolean

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @Test
+    public void testSetAddSessionState() throws Exception {
+        oidcSessionState.setAddSessionState(true);
+        assertNotNull(oidcSessionState.isAddSessionState());
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java]: you can use assertTrue()

**Comment by piraveena** [comment]
Added requested changes

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    private OIDCSessionState oidcSessionState;
+
+    @BeforeTest
+    public void setUp() {
+        oidcSessionState = new OIDCSessionState();
+    }
+
+    @Test
+    public void testSetAuthenticatedUser() {
+        oidcSessionState.setAuthenticatedUser(USERNAME);
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java]: Add line after method signatures

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+/**
+ * Unit test coverage for OIDCSessionParticipantCache class.
+ */
+public class OIDCSessionParticipantCacheTest extends PowerMockTestCase {
+
+    private static final String SESSION_ID = "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
+
+    @Test
+    public void testGetInstance() {
+        assertNotNull(OIDCSessionParticipantCache.getInstance(), "OIDCSessionParticipantCache is null");
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/cache/OIDCSessionParticipantCacheTest.java]: Add line after method signatures

File: components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        assertNotNull(oidcSessionState.getAuthenticatedUser(), "User is not authenticated");
+    }
+
+    @Test
+    public void testSetSessionParticipants() {
+        Set<String> authenticatedUesrs = new HashSet<>();
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionStateTest.java]: Add a new line after the method signature. Fix all the methods.

**Pull request merged**